python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/port-channel-interfaces.j2

```diff
@@ -810,6 +810,126 @@ interface {{ port_channel_interface.name }}
+   switchport tap mpls pop all
+{%         elif port_channel_interface.switchport.tap.encapsulation.vxlan_strip is arista.avd.defined(true) %}
+   switchport tap encapsulation vxlan strip
+{%         endif %}
+{%         for protocol in port_channel_interface.switchport.tap.encapsulation.gre.protocols | arista.avd.natural_sort('protocol') %}
+{%             if protocol.strip is arista.avd.defined(true) %}
+{%                 set tap_encapsulation_cli = "switchport tap encapsulation gre protocol " ~ protocol.protocol %}
+{%                 if protocol.feature_header_length is arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " feature header length " ~ protocol.feature_header_length %}
+{%                 endif %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " strip" %}
+{%                 if protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " re-encapsulation ethernet" %}
+{%                 endif %}
+   {{ tap_encapsulation_cli }}
+{%             endif %}
+{%         endfor %}
+{%         if port_channel_interface.switchport.tap.encapsulation.gre.strip is arista.avd.defined(true) %}
+   switchport tap encapsulation gre strip
+{%         endif %}
+{%         for destination in port_channel_interface.switchport.tap.encapsulation.gre.destinations | arista.avd.natural_sort('destination') %}
+{%             set tap_encapsulation_cli = "switchport tap encapsulation gre destination " ~ destination.destination %}
+{%             if destination.source is arista.avd.defined %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " source " ~ destination.source %}
+{%             endif %}
+{%             for destination_protocol in destination.protocols | arista.avd.natural_sort('protocol') %}
+{%                 if destination_protocol.strip is arista.avd.defined(true) %}
+{%                     set tap_encapsulation_protocol_cli = tap_encapsulation_cli ~ " protocol " ~ destination_protocol.protocol %}
+{%                     if destination_protocol.feature_header_length is arista.avd.defined %}
+{%                         set tap_encapsulation_protocol_cli = tap_encapsulation_protocol_cli ~ " feature header length " ~ destination_protocol.feature_header_length %}
+{%                     endif %}
+{%                     set tap_encapsulation_protocol_cli = tap_encapsulation_protocol_cli ~ " strip" %}
+{%                     if destination_protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
+{%                         set tap_encapsulation_protocol_cli = tap_encapsulation_protocol_cli ~ " re-encapsulation ethernet" %}
+{%                     endif %}
+   {{ tap_encapsulation_protocol_cli }}
+{%                 endif %}
+{%             endfor %}
+{%             if destination.strip is arista.avd.defined(true) %}
```

Comment:
what if i defined the destination and source only. It will render command `"switchport tap encapsulation gre destination " ~ destination.destination ~ " source " ~ destination.source` which is invalid.

Reply:
No, it won't generate the config if `strip` is not defined with `destination` and `source`